File: Codeforces-Sheet-level(0)-mansoura/contest1/problem-C/solution.java
import java.util.Scanner;

public class solution{

    public static void main(String []args){
        Scanner cin = new Scanner(System.in);
        int n = cin.nextInt(); 
        String s = cin.next();
       // char[] arr =s.toCharArray();
        
        int sum_A=0;
        int sum_D=0;
        for(int i=0 ; i< n; i++){
            if(s.charAt(i)=='A') sum_A++;
            if(s.charAt(i)=='D') sum_D++;
        }

        System.out.println(sum_A >sum_D ? "Anton" : sum_D>sum_A? "Danik" :"Frindship");









        cin.close();

    }

}

File: Codeforces-Sheet-level(0)-mansoura/contest1/problem-C/solution.cpp
#include <iostream>


using namespace std;

int main(){

    int n; cin>>n;

    char *arr = new char[n+1];

    int count_A=0;
    int count_D=0;
    for(int i=0; i<n; i++){
        cin>>arr[i];

        if(arr[i]=='A'){
            count_A++;
        } else if(arr[i]=='D'){
            count_D++;
        }
    }
    arr[n] = '\0'; // Null-terminate the string

    cout<< (count_A > count_D ? "Anton" : (count_D > count_A ? "Danik" : "Friendship")) << endl;

     delete []arr;
    
    return 0;
}